src/chartiq_injections/findHighlights.js

```diff
@@ -0,0 +1,543 @@
+            y1: this.valueFromPixel(cy + radius, this.currentPanel),
+            cx0: cx - radius,
+            cx1: cx + radius,
+            cy0: cy - radius,
+            cy1: cy + radius,
+            r: radius,
+        };
+        if (this.repositioningDrawing && box.x1 - box.x0 < 2) {
+            box.x1++;
+            box.x0--;
+        } else if (box.x1 === box.x0) {
+            box.x0 -= 0.5;
+            box.x1 += 0.5;
+        }
+        const markers =
+            this.markerHelper &&
+            this.markerHelper.chartMap[chart.name] &&
+            this.markerHelper.chartMap[chart.name].markers;
+        /* begin test code
+		// show the box
+		this.chart.canvas.context.strokeStyle="red";
+		this.chart.canvas.context.strokeRect(this.pixelFromTick(box.x0,chart),cy-radius,this.pixelFromTick(box.x1,chart)-this.pixelFromTick(box.x0,chart),2*radius);
+		this.chart.canvas.context.strokeStyle="blue";
+		this.chart.canvas.context.strokeRect(cx-radius,cy-radius,2*radius,2*radius);
+		  end test code */
+
+        if (!chart.hideDrawings) {
+            for (let i = this.drawingObjects.length - 1; i >= 0; i--) {
+                const drawing = this.drawingObjects[i];
+                if (!this.panels[drawing.panelName]) continue;
+                if (this.repositioningDrawing && this.repositioningDrawing !== drawing) continue;
+
+                const prevHighlight = drawing.highlighted;
+                let highlightMe = drawing.panelName === this.currentPanel.name;
+                drawing.repositioner = drawing.intersected(this.crosshairTick, this.crosshairValue, box);
+                highlightMe = highlightMe && drawing.repositioner;
+
+                if (!clearOnly && highlightMe) {
+                    if (prevHighlight) {
+                        drawingToMeasure = drawing;
+                        if (this.anyHighlighted && this.singleDrawingHighlight) drawing.highlighted = false;
+                        if (drawing.highlighted && drawing.highlighted !== prevHighlight) somethingChanged = true; // drawing is still highlighted, but a different positioner is active
+                    } else if (prevHighlight !== drawing.highlight(true)) {
+                        if (!drawingToMeasure) drawingToMeasure = drawing;
+                        if (this.anyHighlighted && this.singleDrawingHighlight) drawing.highlighted = false;
+                        somethingChanged = true;
+                    }
+                    this.anyHighlighted = true;
+                } else if (prevHighlight !== drawing.highlight(false)) {
+                        somethingChanged = true;
+                    }
+                if (drawing.highlighted) {
+                    stickyArgs.noDelete = drawing.permanent;
+                    stickyArgs.noEdit = !this.callbackListeners.drawingEdit.length;
+                }
+            }
+        }
+
+        let n, o, m, marker, series, bar;
```

Comment:
Note: had to replace all vars with `let`s & `const`s that's why I declare this `bar` variable here instead of declaring `var bar = ...` at line 307 for the first time.